homeassistant/components/yolink/__init__.py

```diff
@@ -0,0 +1,191 @@
+"""The yolink integration."""
+from __future__ import annotations
+
+import asyncio
+from datetime import timedelta
+import logging
+
+import async_timeout
+import voluptuous as vol
+from yolink.client import YoLinkClient
+from yolink.const import OAUTH2_AUTHORIZE, OAUTH2_TOKEN
+from yolink.device import YoLinkDevice
+from yolink.exception import YoLinkAuthFailError, YoLinkClientError
+from yolink.model import BRDP
+from yolink.mqtt_client import MqttClient
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_CLIENT_ID, CONF_CLIENT_SECRET, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryAuthFailed, ConfigEntryNotReady
+from homeassistant.helpers import (
+    aiohttp_client,
+    config_entry_oauth2_flow,
+    config_validation as cv,
+)
+from homeassistant.helpers.typing import ConfigType
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from . import api, config_flow
+from .const import (
+    ATTR_CLIENT,
+    ATTR_COORDINATOR,
+    ATTR_DEVICE,
+    ATTR_DEVICE_STATE,
+    ATTR_MQTT_CLIENT,
+    DOMAIN,
+)
+
+SCAN_INTERVAL = timedelta(minutes=5)
+
+_LOGGER = logging.getLogger(__name__)
+
+CONFIG_SCHEMA = vol.Schema(
+    {
+        DOMAIN: vol.Schema(
+            {
+                vol.Required(CONF_CLIENT_ID): cv.string,
+                vol.Required(CONF_CLIENT_SECRET): cv.string,
+            }
+        )
+    },
+    extra=vol.ALLOW_EXTRA,
+)
+
+PLATFORMS = [Platform.SENSOR]
+
+
+class YoLinkCoordinator(DataUpdateCoordinator):
+    """YoLink DataUpdateCoordinator."""
+
+    yl_devices: list[YoLinkDevice] = []
```

Comment:
Please don't set mutable objects as class attributes. That's error prone. Move it below to the init method.